plugin/metrics/prometheus/factory.go

```diff
@@ -7,29 +7,29 @@ import (
 	"flag"
 
 	"github.com/spf13/viper"
-	"go.opentelemetry.io/otel"
-	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 
 	"github.com/jaegertracing/jaeger/pkg/prometheus/config"
+	"github.com/jaegertracing/jaeger/pkg/telemetry"
 	"github.com/jaegertracing/jaeger/plugin"
 	prometheusstore "github.com/jaegertracing/jaeger/plugin/metrics/prometheus/metricsstore"
 	"github.com/jaegertracing/jaeger/storage/metricsstore"
+	"github.com/jaegertracing/jaeger/storage/metricsstore/metricstoremetrics"
 )
 
 var _ plugin.Configurable = (*Factory)(nil)
 
 // Factory implements storage.Factory and creates storage components backed by memory store.
 type Factory struct {
 	options *Options
-	logger  *zap.Logger
-	tracer  trace.TracerProvider
+	telset  telemetry.Settings
 }
 
 // NewFactory creates a new Factory.
 func NewFactory() *Factory {
+	telset := telemetry.NoopSettings()
 	return &Factory{
-		tracer:  otel.GetTracerProvider(),
```

Comment:
@yurishkuro is it fine for this tracer to be passed from upstream now?

Reply:
yes. For read path it's generally not a problem which tracer is used.